sdk/python/feast/infra/gcp.py

```diff
@@ -178,3 +146,89 @@ def online_read(
             else:
                 result.append((None, None))
         return result
+
+
+ProtoBatch = Sequence[
+    Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+]
+
+
+def _to_minibatches(data: ProtoBatch, batch_size=50) -> Iterator[ProtoBatch]:
```

Comment:
Should this batch size be closer to (500/operations)? https://stackoverflow.com/questions/52165352/how-can-i-update-more-than-500-docs-in-firestore-using-batch

Reply:
Definitely something that we could experiment more with. I mainly wanted to be on the safer side, and given that there are multiple threads pushing batches, having a smaller batch size gives you a less spiky throughput. Also less chance of transaction conflict with smaller batch so we have to retry less.

Empirically, when pushing from from my laptop to firestore, 50 didn't seem to be worse than 250 in terms of throughput, so I left it there.